### ISSUE TITLE
Backslash should be legal in order to unalias a command passed to xargs

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1196]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1012
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

grep -v foo "${HOME}/*" | xargs \rm -f

\rm -f "$(grep -v foo "${HOME}/*")"


```

#### Here's what shellcheck currently says:
```
Line 3	SC1012: \r is just literal 'r' here. For carriage return, use "$(printf "\r")" instead.
```
#### Here's what I wanted or expected to see:

Nothing. \rm overrides any alias that might be in play, similar to the 'builtin' command for shell builtins, as can be seen from the subsequent line that has no shellcheck complaints. Is its placement as an argument and not a command the issue here?

**Comment by koalaman** [comment]
`xargs` can't invoke aliases anyways, so it is indeed ineffectual here.
